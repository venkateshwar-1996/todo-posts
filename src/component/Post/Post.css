.name-icon {
  height: 100%;
}
.link {
  text-decoration: none !important;
}
.text-off-white {
  color: rgb(0, 0, 0);
}
.tupple {
  border-color: #454d55;
}
.post-by {
  padding-top: 12px;
  padding-left: 20px;
  padding-right: 20px;
}
.post-title {
  padding: 12px 20px;
}
