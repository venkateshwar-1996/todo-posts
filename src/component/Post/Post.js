import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

import "./Post.css";
const Post = ({ userId, userName, id, title }) => {
  return (
    <div data-testid={"post"} className="tupple" >
      <div className="post-by">
        <h5>
          <Link to={`/user/${userId}`}>
            <span className="text-white">{userName}</span>
          </Link>
        </h5>
      </div>
      <Link className="link" to={`/post/${id}`}>
        <div className="post-title">
          <p>
            <span className="text-white">{title}</span>
          </p>
        </div>
      </Link>
    </div>
  );
};

Post.propTypes = {
  userId: PropTypes.number,
  id: PropTypes.number,
  title: PropTypes.string,
  userName: PropTypes.string
};
export default Post;
