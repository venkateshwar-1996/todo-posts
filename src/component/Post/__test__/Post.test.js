import React from "react";
import ReactDOM from "react-dom";
import Post from "../Post";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";

it("Render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <Post userId={1} userName={"Venkatesh"} id={1} title={"React coding"} />
    </MemoryRouter>,
    div
  );
});
it("Render Post correctly", () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <Post userId={1} userName={"Venkatesh"} id={1} title={"React coding"} />
    </MemoryRouter>
  );
  expect(getByTestId('post')).toMatchSnapshot()
});
