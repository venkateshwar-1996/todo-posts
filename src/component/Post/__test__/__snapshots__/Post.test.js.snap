// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Post correctly 1`] = `
<div
  class="tupple"
  data-testid="post"
>
  <div
    class="post-by"
  >
    <h5>
      <a
        href="/user/1"
      >
        <span
          class="text-white"
        >
          Venkatesh
        </span>
      </a>
    </h5>
  </div>
  <a
    class="link"
    href="/post/1"
  >
    <div
      class="post-title"
    >
      <p>
        <span
          class="text-white"
        >
          React coding
        </span>
      </p>
    </div>
  </a>
</div>
`;
